enterprise_catalog/apps/catalog/filters.py

```diff
@@ -0,0 +1,126 @@
+"""
+Utility functions for catalog query filtering without elasticsearch
+"""
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+SUPPORTED_FILTER_COMPARISONS = [
+    'exact',
+    'not',
+    'exclude',
+    'gt',
+    'gte',
+    'lt',
+    'lte',
+]
+
+
+class QueryFilterException(Exception):
+    """
+    An exception for content catalog query filtering
+    """
+
+
+def fix_common_query_key_mistakes(raw_query_key):
+    """
+    In production many queries have odd typos
+    which seem to have been copypasta-proliferated
+    """
+    raw_query_key = raw_query_key.lower()
+    if raw_query_key == 'aggregration__key':
+        return 'aggregation_key'
+    elif raw_query_key == 'aggregation__key':
+        return 'aggregation_key'
+    elif raw_query_key == 'org__exempt':
+        return 'org__exclude'
+    return raw_query_key
+
+
+def extract_field_and_comparison_kind(raw_query_key):
+    """
+    Taking an query key, extra the content_metadata
+    field name and the kind of comparison matching
+    should be used.
+    """
+    field = None
+    # comparison_kind defaults to "exact match"
+    comparison_kind = 'exact'
+    if len(raw_query_key.split("__")) == 2:
```

Comment:
`raw_query_key.split("__")` expression gets duplicated a couple times
```suggestion
    field__comparison = raw_query_key.split("__")
    if len(field__comparison) == 2:
```